### ISSUE TITLE
IDISP001 should not warn when returning local in lambda

### ISSUE THREAD

**Issue by JohanLarsson** [DotNetAnalyzers/IDisposableAnalyzers#70]
```cs
namespace RoslynSandbox
{
    using System;

    public sealed class Foo : IDisposable
    {
        private readonly Lazy<IDisposable> disposable;

        public Foo()
        {
            this.disposable = new Lazy<IDisposable>(() =>
            {
                var temp = new Disposable();
                return temp;
            });
        }

        public void Dispose()
        {
            if (this.disposable.IsValueCreated)
            {
                this.disposable.Value.Dispose();
            }
        }
    }
}
```
